File: app/src/main/java/com/example/barbershop/Api/services/ApiServices.java
package com.example.barbershop.Api.services;

import com.example.barbershop.Api.entity.Users;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiServices {
    @GET("api/v1/user/{id}")
    Call<Users> getId(@Path("id") int id);

    @GET("api/v1/usuarios")
    Call<List<Users>> getUserId(@Query("userId") int userId);

    @GET("api/v1/usuarios")
    Call<List<Users>> getUserById(@Query("id") int id);
}

File: app/src/main/java/com/example/barbershop/Api/services/ApiServices.kt
package com.example.barbershop.Api.services

import com.example.barbershop.Api.entity.User
import retrofit2.Call
import retrofit2.http.GET

interface ApiServices {
    @GET("api/v1/usuarios")
    fun getIdUser(): Call<User>

//    @GET("api/v1/usuarios")
//    fun getId(): Response<ApiResponse<List<Users>>>

//    @GET("api/v1/usuarios")
//    fun getUserId(@Query("IdUser") userId: Int): Call<List<Users?>?>?
//
//    @GET("api/v1/usuarios")
//    fun getUserById(@Query("id") id: Int): Call<List<Users?>?>?
}